src/nvim/api/private/helpers.h

```diff
@@ -8,15 +8,10 @@
 #include "nvim/memory.h"
 #include "nvim/lib/kvec.h"
 
+void _api_set_error(Error *err, ErrorType errType, const char *format, ...);
 // -V:api_set_error:618
 #define api_set_error(err, errtype, ...) \
-  do { \
-    snprintf((err)->msg, \
-             sizeof((err)->msg), \
-             __VA_ARGS__); \
-    (err)->set = true; \
-    (err)->type = kErrorType##errtype; \
-  } while (0)
+  _api_set_error(err, kErrorType##errtype, __VA_ARGS__)
```

Comment:
Though I do not see much sense in the macro any longer: it only hides `kErrorType` prefix, not something I would consider good under this circumstances (when passing e.g. Validation to a function you would not expect that it is not a full identifier, but a part of it: all-lower does not look like a macro; with all-caps this is slightly better but I do not think ten characters are worth it).